.github/workflows/pull_request_verify.yaml

```diff
@@ -0,0 +1,133 @@
+name: Pull Requests
+
+on:
+  pull_request:
+    types: [ opened, synchronize, reopened ]
+
+env:
+  CARGO_TERM_COLOR: always
+  RUST_BACKTRACE: full
+  CARGO_TARGET_DIR: ${{ github.workspace }}/../target
+  LOGBAK_USER: ${{secrets.LOGBAK_USER}}  #LOCBAK_* for upload logs to server when test failed, for windows
+  LOGBAK_PASSWORD: ${{secrets.LOGBAK_PASSWORD}}
+  LOGBAK_SERVER: ${{secrets.LOGBAK_SERVER}}
+jobs:
+  checkout:
+    name: ckb-checkout
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [Linux-runner, mac-runner,win-runner]
+    steps:
+    - uses: actions/checkout@v2
+
+  UnitTest:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [Linux-runner, mac-runner,win-runner]
+        include:
+          - os: Linux-runner
+            script_run: make test
+          - os:  mac-runner
+            script_run: make test
+          - os: win-runner
+            script_run: devtools/windows/make test
+            CI: true
+            ImageOS: 'windows-2019'
+            BUILD_BUILDID: ${{ github.sha }}
+    needs: checkout
```

Comment:
Notice that you depend on the "checkout" job by `needs: checkout`. 
Could we just check out the repository on the first step of a job by using `- uses: actions/checkout@v2`? And what are the differences?

```
env:
  CI: ${{ matrix.CI }}
  ImageOS:  ${{ matrix.ImageOS }}
steps:
  - uses: actions/checkout@v2
  - run: ${{ matrix.script_run }}
```

Reply:
My opinion is define the job name and check logs clearly.
Let me have a try, will modify it if working check out the repository on the first step of a job by using - uses: actions/checkout@v2.